site/docs/sql/intro.md

```diff
@@ -0,0 +1,306 @@
+
+- SQL queries over [Apache Kafka topics](kafka-connector.md) and files
+(local and remote) (TODO)
+- Joining Kafka or file data with local IMaps (enrichment)
+- Filtering and projection using [SQL
+expressions](https://docs.hazelcast.org/docs/{imdg-version}/manual/html-single/index.html#expressions)
+- Aggregating data from files using predefined [aggregate
+functions](basic-commands#aggregation-functions)
+- Receiving query results via Jet client (Java) or writing the results
+to an [IMap](imap-connector.md) in the Jet cluster
+- Running continuous (streaming) and batch queries, see
+[Job Management](job-management.md)
+
+These are some of the features on our roadmap:
+
+- Joins with arbitrary external data sources
+- Windowed aggregation of streaming data
+- JDBC
+
+## Installation
+
+You need the `hazelcast-jet-sql` module on your classpath. For
+Gradle or Maven, make sure to add the dependency:
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--Gradle-->
+
+```groovy
+compile 'com.hazelcast.jet:hazelcast-jet-sql:{jet-version}'
+```
+
+<!--Maven-->
+
+```xml
+<dependency>
+    <groupId>com.hazelcast.jet</groupId>
+    <artifactId>hazelcast-jet-sql</artifactId>
+    <version>{jet-version}</version>
+</dependency>
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+If you're using the distribution package, make sure to move the
+`hazelcast-jet-sql-{jet-version}.jar` file from the `opt/` to the `lib/`
+directory.
+
+If you use other features from SQL such as the Kafka or File connectors,
+also add the `hazelcast-jet-kafka` or `hazelcast-jet-hadoop` modules in
+the same way.
+
+## Example
+
+In this example we'll show how to query Apache Kafka using SQL and
+stream the messages to an IMap.
+
+First, make sure you have Apache Kafka up and running. Please follow any
+of the Kafka tutorials. Start a broker and create a topic named
+`trades`.
+
+The trades will be encoded as JSON messages:
+
+```plain
+key: ABCD
+value: {
+    "ticker": "ABCD",
+    "price": 5.5,
+    "amount": 10
+}
+```
+
+The `ticker` is repeated in both the key and the value. We'll ignore the
+key in the subsequent examples, it's only used by Kafka for
+partitioning.
+
+### Creating the Mapping for Kafka Topic
+
+To use a remote topic as a table in Jet, first create an `EXTERNAL
+MAPPING` for the topic. It maps the messages to a fixed list of columns
+with data types:
+
+```sql
+CREATE MAPPING trades (
+    ticker VARCHAR,
+    price DECIMAL,
+    amount BIGINT)
+TYPE Kafka
+OPTIONS (
+    valueFormat 'json',
+    "bootstrap.servers" '1.2.3.4:9092'
+    /* ... more configuration options for the Kafka consumer */
+)
+```
+
+The `valueFormat` option specifies the serialization format for the
+value. For other possible values see the [Kafka
+Connector](kafka-connector) page. The options not handled by Jet are all
+passed to the Kafka consumer or producer, such as the
+`bootstrap.servers` above.
+
+To submit the above query, use the Java API (we plan JDBC support and
```

Comment:
We plan CLI at the first place. Please mention it.

Reply:
The current draft of CLI doesn't support multi-line commands. I'll add a todo to add it later.